### PR TITLE
Issue 1101 load transfer fn options

### PR DESCRIPTION
**Pull request by sgpearse**
Added options to load .tf3 values for data range and opacity.

### CODE REVIEW

**Comment by sgpearse** [comment]
@clyne 

The bounds issue is fixed, along with another opacity bug.  However I'm not sure why the directory path is failing.  It works for myself and Stas.  I'm using the following to get the path:

From TFWidget.cpp:
1040     QString directory = QString::fromStdString(
1041         Wasp::GetSharePath( string("palettes") )
1042     );
1043     _fileDialog->setDirectory(directory);

**Comment by sgpearse** [comment]
Closing temporarily while troubleshooting a new bug.

**Comment by clyne** [comment]
Did this PR disappear? 

> On Dec 19, 2018, at 3:36 PM, sgpearse <notifications@github.com> wrote:
> 
> @clyne
> 
> The bounds issue is fixed, along with another opacity bug. However I'm not sure why the directory path is failing. It works for myself and Stas. I'm using the following to get the path:
> 
> From TFWidget.cpp:
> 1040 QString directory = QString::fromStdString(
> 1041 Wasp::GetSharePath( string("palettes") )
> 1042 );
> 1043 _fileDialog->setDirectory(directory);
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
> 

John Clyne
National Center for Atmospheric Research
303.497.1236 (w), 303.809.1922 (c)
clyne@ucar.edu

**Comment by sgpearse** [comment]
@clyne 

I had a memory issue involving Qt's parent/child destructor policy.  It's fixed now.

It's hard to see, but I commented about the problem yesterday before closing it.  I didn't want anybody to merge it until this was resolved.

File: apps/vaporgui/TFWidget.cpp [hunk 1]
```diff
@@ -133,69 +142,94 @@ void TFWidget::Reinit(TFFlags flags) {
...
+    for (int i=0; i<numOpacityMaps; i++) {
+        controlPoints.push_back(tf->GetOpacityMap(i)->GetControlPoints());
+    }
 
-	int rc = tf->LoadFromFile(s.toStdString(), defaultRange);
+	_paramsMgr->BeginSaveStateGroup("Loading Transfer Function from file");
+ 
+	rc = tf->LoadFromFile(fileName);
 	if (rc<0) {
 		MSG_ERR("Error loading transfer function");
```
Comment (clyne) [inline apps/vaporgui/TFWidget.cpp]: You are returning from within a "StateSaveGroup". You need to call paramsMgr->EndSaveStateGroup() or you'll mess up the session state

**Pull request merged**